include/riak_kv_ts.hrl

```diff
@@ -29,12 +29,12 @@
 %% For dialyzer types
 -include_lib("riak_ql/include/riak_ql_ddl.hrl").
 
--define(SQL_SELECT, #riak_select_v1).
--define(SQL_SELECT_RECORD_NAME, riak_select_v1).
+-define(SQL_SELECT, #riak_select_v2).
+-define(SQL_SELECT_RECORD_NAME, riak_select_v2).
 
 %% the result type of a query, rows means to return all matching rows, aggregate
 %% returns one row calculated from the result set for the query.
--type select_result_type() :: rows | aggregate.
+-type select_result_type() :: rows | aggregate | group_by.
```

Comment:
There is an interesting question about changing the type of a record - that presumably implies that there are changes the data passed in that record that might make the record incompatible over the wire.

This would imply bumping #riak_sel_clause_v1 to #riak_sel_clause_v2 instead of extending the type. 

I suspect that this is the case here - but it is being covered by the fact that we have bumped the version of the record that contains this record...

Should we have a general policy on 'upgrades in place' in records by type extension? Not sure this question can/should be fixed/addressed in this PR - but raising it anyhoo...